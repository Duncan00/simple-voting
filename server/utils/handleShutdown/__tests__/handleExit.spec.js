'use strict';

const handleExit = require('../handleExit');

describe('handleExit', () => {
	test('should gracefully shutdown', async () => {
		const app = {
			isTerminating: false
		};
		const redis = {
			quit: jest.fn()
		};
		const logger = {
			info: jest.fn()
		};
		const server = {
			close: jest.fn().mockImplementation(async callback => {
				await callback();
			})
		};
		process.exit = jest.fn();

		await handleExit({server, app, redis, logger})();

		expect(redis.quit.mock.calls).toMatchSnapshot('redis.quit');
		expect(logger.info.mock.calls).toMatchSnapshot('logger.info');
		expect(logger.info.mock.calls).toMatchSnapshot('logger.info');
		expect(process.exit.mock.calls).toMatchSnapshot('process.exit');
	});

	test('should skip if already shutting down', async () => {
		const app = {
			isTerminating: true
		};
		const redis = {
			quit: jest.fn()
		};
		const logger = {
			info: jest.fn()
		};
		const server = {
			close: jest.fn().mockImplementation(async callback => {
				await callback();
			})
		};
		process.exit = jest.fn();

		await handleExit({server, app, redis, logger})();

		expect(redis.quit.mock.calls).toMatchSnapshot('redis.quit');
		expect(logger.info.mock.calls).toMatchSnapshot('logger.info');
		expect(logger.info.mock.calls).toMatchSnapshot('logger.info');
		expect(process.exit.mock.calls).toMatchSnapshot('process.exit');
	});
});
