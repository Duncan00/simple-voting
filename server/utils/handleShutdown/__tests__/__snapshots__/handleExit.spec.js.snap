// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleExit should gracefully shutdown: logger.info 1`] = `
Array [
  Array [
    "Signal received application starts shutdown",
  ],
  Array [
    "Application had gracefully shutdown",
  ],
]
`;

exports[`handleExit should gracefully shutdown: logger.info 2`] = `
Array [
  Array [
    "Signal received application starts shutdown",
  ],
  Array [
    "Application had gracefully shutdown",
  ],
]
`;

exports[`handleExit should gracefully shutdown: process.exit 1`] = `
Array [
  Array [],
]
`;

exports[`handleExit should gracefully shutdown: redis.quit 1`] = `
Array [
  Array [],
]
`;

exports[`handleExit should skip if already shutting down: logger.info 1`] = `
Array [
  Array [
    "Signal received application starts shutdown",
  ],
]
`;

exports[`handleExit should skip if already shutting down: logger.info 2`] = `
Array [
  Array [
    "Signal received application starts shutdown",
  ],
]
`;

exports[`handleExit should skip if already shutting down: process.exit 1`] = `Array []`;

exports[`handleExit should skip if already shutting down: redis.quit 1`] = `Array []`;
